crossbar/worker/router.py

```diff
@@ -1055,6 +1056,57 @@ def create_resource(self, path_config):
             #
             return CallerResource(path_config.get('options', {}), caller_session)
 
+        # File Upload resource
+        #
+        elif path_config['type'] == 'fileupload':
+
+            fileupload_directory = os.path.abspath(os.path.join(self.config.extra.cbdir, path_config['directory']))
+            fileupload_directory = fileupload_directory.encode('ascii', 'ignore')  # http://stackoverflow.com/a/20433918/884770
+            
+
+            temp_dir = os.path.abspath(os.path.join(self.config.extra.cbdir, path_config['temp_directory']))
+            temp_dir = temp_dir.encode('ascii', 'ignore')  # http://stackoverflow.com/a/20433918/884770
+
+            max_file_size = 1024 * 1024 * 1000   # 1000MB
+            if 'max_file_size' in path_config:  
+                max_file_size = path_config['max_file_size']
+
+            form_fields = path_config['form_fields']
+                
+            mime_types = ''
+            if 'mime_types' in path_config:  
+                mime_types = path_config['mime_types']
+            
+            file_types = []
+            if 'file_types' in path_config:  
+                file_types = path_config['file_types']
+
+            file_owner = {}
+            if 'file_owner' in path_config: 
+                file_owner = path_config['file_owner'] 
+
+            file_progress_URI = ''
+            # If fileupload events are not desired the publish function does nothing then.
+            def fileupload_publish(payload):
+                return ''
+
+            if 'file_progress' in path_config:  
+                file_progress_URI = path_config['file_progress']['uri']
+
+                fileupload_session_config = ComponentConfig(realm=path_config['file_progress']['realm'], extra=None)
```

Comment:
ok with ownership, but I can change the permissions. I'll only keep the permission config then.
And yes, we should definitely get the config right.